optimum/onnxruntime/modeling_ort.py

```diff
@@ -0,0 +1,681 @@
+    def _from_pretrained(
+        cls,
+        model_id: Union[str, Path],
+        use_auth_token: Optional[Union[bool, str, None]] = None,
+        revision: Optional[Union[str, None]] = None,
+        force_download: bool = True,
+        cache_dir: Optional[str] = None,
+        file_name: Optional[str] = None,
+        **kwargs,
+    ):
+        """
+        load a model and its configuration file from a directory or the HF Hub.
+        Implements: https://github.com/huggingface/huggingface_hub/blob/e67de48368bc1843e40afc1cc9d236402b9609ee/src/huggingface_hub/hub_mixin.py#L73
+        Arguments:
+            model_id (:obj:`str` or :obj:`Path`):
+                Directory from which to load
+            use_auth_token (:obj:`str` or :obj:`bool`):
+                Is needed to load models from a private repository
+            revision (:obj:`str`):
+                Revision is the specific model version to use. It can be a branch name, a tag name, or a commit id
+            cache_dir (:obj:`Union[str, Path]`, `optional`):
+                Path to a directory in which a downloaded pretrained model configuration should be cached if the
+                standard cache should not be used.
+            force_download (:obj:`bool`, `optional`, defaults to :obj:`False`):
+                Whether or not to force the (re-)download of the model weights and configuration files, overriding the
+                cached versions if they exist.
+            file_name(:obj:`str`):
+                Overwrites the default model file name from `"model.onnx"` to `file_name`. This allows you to load different model files from the same
+                repository or directory.
+            kwargs (:obj:`Dict`, `optional`)::
+                kwargs will be passed to the model during initialization
+        """
+        config_dict = kwargs.pop("config", {})
+        model_file_name = file_name if file_name is not None else ONNX_WEIGHTS_NAME
+        # load model from local directory
+        if os.path.isdir(model_id):
+            config = PretrainedConfig.from_dict(config_dict)
+            model = OnnxModel.load_model(os.path.join(model_id, model_file_name))
+            kwargs["model_save_dir"] = Path(model_id)
+        # load model from hub
+        else:
+            # download model
+            model_cache_path = hf_hub_download(
+                repo_id=model_id,
+                filename=model_file_name,
+                use_auth_token=use_auth_token,
+                revision=revision,
+                cache_dir=cache_dir,
+                force_download=force_download,
+            )
+            kwargs["model_save_dir"] = Path(model_cache_path).parent
+            kwargs["latest_model_name"] = Path(model_cache_path).name
+            model = OnnxModel.load_model(model_cache_path)
+            config = PretrainedConfig.from_dict(config_dict)
+        return cls(model=model, config=config, **kwargs)
+
+    @classmethod
+    def from_transformers(
```

Comment:
Cf comment for the optimize method.
Again, I think this is neat, but I'm not sure we should have all of those features embedded inside the `OnnxModel` class. Just want to open the discussion with the others.

Reply:
Happy to discuss this tomorrow and then comment on the result of it here. I think we should keep the `.from_transformers` method since it is creating the bridge between `transformers` and `optimum` providing the API `transformers` users are familiar with and abstracting all optimization away. 
I added tests to make sure that all supported exports are working that way we know what works and what doesn't.